### ISSUE TITLE
Bug: [calcite-label] produces duplicate ID elements on the page

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-components#1106]
### Summary

Setting ID on a calcite-label will cause duplicate ID elements. Setting this as high priority as this causes a11y audits to fail (https://dequeuniversity.com/rules/axe/4.0/duplicate-id?application=axeAPI).

### Actual Behavior

The internal label gets the host's `id` attribute passed through and exposed to the page since the component does not use shadow DOM.

### Expected Behavior

The internal label ID should not be exposed on the page.

### Reproduction Steps

The following codepen illustrates the issue: https://codepen.io/jcfranco-the-scripter/pen/mdEygWa?editors=1010
